nix/mobile/android/targets/release-android.nix

```diff
@@ -0,0 +1,129 @@
+      inherit path;
+      name = "status-react-${name}-source";
+      filter =
+        # Keep this filter as restrictive as possible in order to avoid unnecessary rebuilds and limit closure size
+        mkFilter {
+          dirRootsToInclude = [ 
+            "mobile_files"
+            "modules/react-native-status"
+            "packager"
+            "resources"
+          ];
+          dirsToExclude = [ ".git" ".svn" "CVS" ".hg" ".gradle" "build" "intermediates" "libs" "obj" ];
+          filesToInclude = [ envFileName "STATUS_GO_VERSION" "VERSION" ];
+          root = path;
+        };
+    };
+  nativeBuildInputs = [ bash gradle ] ++ lib.optionals stdenv.isDarwin [ file gnumake watchman ];
+  buildInputs = [ nodejs openjdk ];
+  phases = [ "unpackPhase" "patchPhase" "buildPhase" "installPhase" ];
+  postUnpack = ''
+    mkdir -p ${gradleHome}
+
+    ${if keystore-file != "" then "cp -a --no-preserve=ownership ${keystore-file} ${gradleHome}/; export STATUS_RELEASE_STORE_FILE=${gradleHome}/$(basename ${keystore-file})" else ""}
+
+    # Ensure we have the right .env file
+    cp -f $sourceRoot/${envFileName} $sourceRoot/.env
+
+    # Copy index.*.js input file
+    cp -a --no-preserve=ownership ${prod-build}/index*.js $sourceRoot/
+
+    # Copy android/ directory
+    cp -a --no-preserve=ownership ${sourceProjectDir}/android/ $sourceRoot/
+    chmod u+w $sourceRoot/android
+    chmod u+w $sourceRoot/android/app
+    chmod -R u+w $sourceRoot/android/.gradle
+    mkdir $sourceRoot/android/build && chmod -R u+w $sourceRoot/android/build
+
+    # Copy node_modules/ directory
+    #ln -sf ${sourceProjectDir}/node_modules/
+
+    # Make android/build directories writable under node_modules
+    cp -a --no-preserve=ownership ${sourceProjectDir}/node_modules/ $sourceRoot/
+    chmod -R u+w $sourceRoot/node_modules/react-native/
+    for d in `ls $sourceRoot/node_modules/react-native-*/android/build -d1`; do
+      chmod -R u+w $d
+    done
+    for d in `ls $sourceRoot/node_modules/react-native-*/android -d1`; do
+      chmod u+w $d
+    done
+    chmod u+w $sourceRoot/node_modules/realm/android
+  '';
+  patchPhase = ''
+    prevSet=$-
+    set -e
+
+    substituteInPlace android/gradlew \
+      --replace 'exec gradle' "exec gradle -S -Dmaven.repo.local='${localMavenRepo}' --offline ${gradle-opts}"
+    substituteInPlace android/app/build.gradle \
+      --replace 'versionCode 9999' 'versionCode ${build-number}'
+
+    # OPTIONAL: There's no need to forward debug ports for a release build, just disable it
+    substituteInPlace node_modules/realm/android/build.gradle \
+      --replace 'compileTask.dependsOn forwardDebugPort' 'compileTask'
+
+    set $prevSet
+  '';
+  buildPhase =
+    let
+      exportEnvVars = lib.concatStrings (
+        builtins.attrValues (
+          lib.mapAttrs (name: value:
+            "export ${name}='${value}';"
+            ) env));
+      unsetEnvVars = lib.concatStrings (
```

Comment:
Can you explain these `lib.concatStrings` here with comment? I'm at a loss on what they do.

Reply:
I'm simplifying this code as follows:

```nix
      # Take the env attribute set and build a couple of scripts
      #  (one to export the environment variables, and another to unset them)
      exportEnvVars = concatStringsSep ";" (mapAttrsToList (name: value: "export ${name}='${value}'") env);
      unsetEnvVars = concatStringsSep ";" (mapAttrsToList (name: value: "unset ${name}") env);
```